moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -570,8 +570,11 @@ bool JogCalcs::checkIfJointsWithinURDFBounds(trajectory_msgs::JointTrajectory& n
       {
         if (new_joint_traj.joint_names[c] == joint->getName())
         {
-          new_joint_traj.points[0].velocities[c] = kinematic_state_->getJointVelocities(joint)[0];
-          break;
+          if ((new_joint_traj.points.size() > 0) && (new_joint_traj.points[0].velocities.size() > c + 1))
+          {
+            new_joint_traj.points[0].velocities[c] = *(kinematic_state_->getJointVelocities(joint));
+            break;
+          }
```

Comment:
Does this need an ``else error`` ?

Reply:
It seems odd to me the ``joint_model_group_`` and ``new_joint_traj`` both would have a joint that does not have velocity initialized, while the rest do. This seems like undesired behavior.

Your fix seems ok for now, but please add TODO raising this question of how this would happen